soc/arm/unisoc_uwp/uwp566x/soc.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2018, UNISOC Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+
+#include <uwp_hal.h>
+
+extern void uwp_cache_init(void);
```

Comment:
Why the extern's if we aren't using / calling these functions?

Reply:
Move these extern functions to header file.